thirdparty/PxPay.php

```diff
@@ -37,43 +37,29 @@ public function makeRequest($request)
         $request->setKey($this->PxPay_Key);
         $xml = $request->toXml();
 
-        // parsing the given URL
-        $URL_Info = parse_url($this->PxPay_Url);
+        return $this->submitXml($xml);
+    }
 
-        // Building referrer
-        if (!isset($referrer) || $referrer == "") // if not given use this script as referrer
-        {
-            $referrer = $_SERVER["HTTP_REFERER"];
-        }
+    /**
+     * Actual submission of XML using cURL. Returns output XML
+     * @param string $inputXml
+     * @return string
+     */
+    public function submitXml($inputXml)
+    {
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $this->PxPay_Url);
 
-        // Find out which port is needed - if not given use standard (=80)
-        if (!isset($URL_Info["port"])) {
-            $URL_Info["port"] = 443;
-        }
-        #	$URL_Info["port"]=80;
-
-        // building POST-request:
-        $requestdata = "POST " . $URL_Info["path"] . " HTTP/1.1\n";
-        $requestdata .= "Host: " . $URL_Info["host"] . "\n";
-        #   $requestdata.="User-Agent: PHP Script\n";
-        $requestdata .= "Referer: $referrer\n";
-        $requestdata .= "Content-type: application/x-www-form-urlencoded\n";
```

Comment:
🏕️ Should we be sending the Content-type header with the new cURL request?

Reply:
Hi @vinstah good observation, and I think we don't need to since the curl implementation is based on the latest example code (version 2) from Windcave